backend/src/modules/tournament-tree/core/domain/FightingTeamsGroup.ts

```diff
@@ -7,19 +7,22 @@ export class FightingTeamsGroup {
   secondTeam: TournamentTeam | undefined;
   readonly fightingTeamsGroupLevel: number;
   nextMatchId: FightingTeamsGroupId | undefined;
+  matchNumberInSequence: number | undefined;
```

Comment:
W jakim przypadku to moze byc undefined? Bo w testach nie widzę czegoś takiego. Rozumiem, ze w meczu moze nie byc druzyn - bo jeszczen nie ustalone. Moze nie byc nextMatchId, bo to final. Ale jak moze mecz nie miec numeru :) ?

Reply:
Ponieważ kolejność meczu jest nadawana na późniejszym etapie - na końcowym. 
Algorytm będzie inny w nadawaniu wartości tej zmiennej dla drzewka pojedyńczej i podwójnej eliminacji.
Dlatego początkowo wydaje się, że dobrze jak bedzie undefined, ale po utworzeniu drzewka już powinna być dodana.

Jesli masz lepsze rozwiązanie to pisz - nie mówię, ze to najlepsze!

Reply:
W ogóle teraz bym nie patrzył na drzewka pojedynczej i podwójnej eliminacji. Jeśli tylko po to jest `number | undefined` to w takim razie to drzewko cierpi na rozdwojenie jaźni :) Nie wie, czy jest drzewkiem pojedynczym, czy podwójnym. Lepiej wtedy zrobić inną implementację specjalnie dla danego drzewka.